autopts/utils.py

```diff
@@ -477,6 +479,61 @@ def log_memory_usage():
     logging.debug(f"Memory usage: {mem_usage:.2f} MB")
 
 
+def extract_wid_testcases_to_csv():
+    log_dir = "./logs"
+    output_csv_path = "./wid_usage_report.csv"
+    profile_wid_map = defaultdict(lambda: defaultdict(set))
+
+    for root, _, files in os.walk(log_dir):
+        for filename in files:
+            if filename.endswith(".log"):
+                filepath = os.path.join(root, filename)
+                with open(filepath, encoding='utf-8', errors='ignore') as f:
+                    lines = f.readlines()
+
+                in_block = False
+                wid = None
+                test_case_name = None
+
+                for line in lines:
+                    stripped = line.strip()
+
+                    if stripped.startswith("BEGIN OnImplicitSend:"):
+                        in_block = True
+                        wid = None
+                        test_case_name = None
```

Comment:
Should this be

```suggestion
                    if not in_block and stripped.startswith("BEGIN OnImplicitSend:"):
                        in_block = True
                        wid = None
                        test_case_name = None
```

Otherwise isn't there a small risk that `wid` and `test_case_name` may be overwritten if `lines` for some reason contains multiple `BEGIN OnImplicitSend`?